### PR TITLE
Feature/cold warm for vector models

### PR DESCRIPTION
**Pull request by feldlime**
## Description

- Added `dataset` argument to `recommend_for_cold` method
- Supported warm and cold reco in `LightFMWrapperModel`
- Supported warm reco in `DSSMModels` and in corresponding pytorch datasets
- [Breaking] Replaced `include_warm` parameter in `Dataset.get_user_item_matrix` to pair `include_warm_users` and `include_warm_items`
- [Breaking] Added `n_factors` and `deterministic` params to `DSSMModel`
- [Breaking] Added default value for train dataset type and params for user and item dataset types in `DSSMModel` 
- [Breaking] Replaced `include_warm` parameter in `Dataset.get_user_item_matrix` to pair `include_warm_users` and `include_warm_items` 
- [Breaking] Renamed torch datasets and `dataset_type` to `train_dataset_type` param in `DSSMModel` 

Closes https://github.com/MobileTeleSystems/RecTools/issues/121

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Optimization

## How Has This Been Tested?
Before submitting a PR, please check yourself against the following list. It would save us quite a lot of time.
- Have you read the contribution guide?
- Have you updated the relevant docstrings? We're using Numpy format, please double-check yourself
- Does your change require any new tests?
- Have you updated the changelog file?

### CODE REVIEW

**Review by blondered — DISMISSED** [review]

File: rectools/models/dssm.py [hunk 1]
```diff
@@ -261,12 +274,15 @@ def __init__(
...
         verbose: int = 0,
+        deterministic: bool = False,
     ) -> None:
         super().__init__(verbose=verbose)
         self.model: tp.Optional[DSSM]
         self._model = model
+        self.n_factors = n_factors
         self.max_epochs = max_epochs
         self.batch_size = batch_size
+        self.deterministic = deterministic
```
[the remark this answers is not in the document]
Comment (feldlime) [reply rectools/models/dssm.py]: removed

File: rectools/models/dssm.py [hunk 2]
```diff
@@ -262,7 +268,9 @@ class DSSMModel(VectorModel):
 
     def __init__(
         self,
-        dataset_type: TorchDataset[tp.Any],
+        train_dataset_type: tp.Type[DSSMTrainDatasetBase] = DSSMTrainDataset,
```
Comment (blondered) [inline rectools/models/dssm.py:271]: You also need to updated docstrings

**Review by blondered — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,12 +10,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Warm and cold users/items support in `cross_validate` ([#77](https://github.com/MobileTeleSystems/RecTools/pull/77))
+- Default value for train dataset type  and params for user and item dataset types in `DSSMModel` ([#122](https://github.com/MobileTeleSystems/RecTools/pull/122))
 
 ### Changed
 - Changed the logic of choosing random sampler for `RandomModel` and increased the sampling speed ([#120](https://github.com/MobileTeleSystems/RecTools/pull/120))
 - Changed the logic of `RandomModel`: now the recommendations are different for repeated calls of recommend methods ([#120](https://github.com/MobileTeleSystems/RecTools/pull/120))
+- Torch datasets to support warm recommendations ([#122](https://github.com/MobileTeleSystems/RecTools/pull/122))
+- Replaced `include_warm` parameter in `Dataset.get_user_item_matrix` to pair `include_warm_users` and `include_warm_items` ([#122](https://github.com/MobileTeleSystems/RecTools/pull/122))
+- Renamed torch datasets and `dataset_type` to `train_dataset_type` param in `DSSMModel` ([#122](https://github.com/MobileTeleSystems/RecTools/pull/122))
```
Comment (blondered) [inline CHANGELOG.md:22]: This is a breaking change. Old interface won't work. Should we write this in changelog so that we don't miss it in release notes?

Comment (blondered) [reply CHANGELOG.md:22]: Also please add this to a PR description

File: rectools/models/dssm.py (hunk 2 above)
Comment (feldlime) [reply rectools/models/dssm.py:280]: done

File: CHANGELOG.md (hunk 3 above)
Comment (feldlime) [reply CHANGELOG.md]: done

**Review by blondered — APPROVED** [review]

**Pull request merged**